packages/wrangler/src/r2/helpers.ts

```diff
@@ -792,6 +792,165 @@ export async function updateR2DevDomain(
+		return `on ${displayDate}`;
+	}
+
+	return "";
+}
+
+export function tableFromLifecycleRulesResponse(rules: LifecycleRule[]): {
+	id: string;
+	enabled: string;
+	prefix: string;
+	action: string;
+}[] {
+	const rows = [];
+	for (const rule of rules) {
+		const actions = [];
+
+		if (rule.deleteObjectsTransition) {
+			const action = "Expire objects";
+			const condition = formatCondition(rule.deleteObjectsTransition.condition);
+			actions.push(`${action} ${condition}`);
+		}
+		if (
+			rule.storageClassTransitions &&
+			rule.storageClassTransitions.length > 0
+		) {
+			for (const transition of rule.storageClassTransitions) {
+				const action = "Transition to Infrequent Access";
+				const condition = formatCondition(transition.condition);
+				actions.push(`${action} ${condition}`);
+			}
+		}
+		if (rule.abortMultipartUploadsTransition) {
+			const action = "Abort incomplete multipart uploads";
+			const condition = formatCondition(
+				rule.abortMultipartUploadsTransition.condition
+			);
+			actions.push(`${action} ${condition}`);
+		}
+
+		rows.push({
+			id: rule.id,
+			enabled: rule.enabled ? "Yes" : "No",
+			prefix: rule.conditions.prefix || "(all prefixes)",
+			action: actions.join(", ") || "(none)",
+		});
+	}
+	return rows;
+}
+
+export async function getLifecycleRules(
+	accountId: string,
+	bucket: string,
+	jurisdiction?: string
+): Promise<LifecycleRule[]> {
+	const headers: HeadersInit = {};
+	if (jurisdiction) {
+		headers["cf-r2-jurisdiction"] = jurisdiction;
+	}
+
+	const result = await fetchResult<{ rules: LifecycleRule[] }>(
+		`/accounts/${accountId}/r2/buckets/${bucket}/lifecycle`,
+		{
+			method: "GET",
+			headers,
+		}
+	);
+	return result.rules;
+}
+
+export async function putLifecycleRules(
+	accountId: string,
+	bucket: string,
+	rules: LifecycleRule[],
+	jurisdiction?: string
+): Promise<void> {
+	const headers: HeadersInit = {
+		"Content-Type": "application/json",
+	};
+	if (jurisdiction) {
+		headers["cf-r2-jurisdiction"] = jurisdiction;
+	}
+
+	await fetchResult(`/accounts/${accountId}/r2/buckets/${bucket}/lifecycle`, {
+		method: "PUT",
+		headers,
+		body: JSON.stringify({ rules: rules }),
+	});
+}
+
+export function formatActionDescription(action: string): string {
+	switch (action) {
+		case "expire":
+			return "expire objects";
+		case "transition":
+			return "transition to Infrequent Access storage class";
+		case "abort-multipart":
+			return "abort incomplete multipart uploads";
+		default:
+			return action;
+	}
+}
+
+export function isValidDate(dateString: string): boolean {
+	const regex = /^\d{4}-\d{2}-\d{2}$/;
+	if (!regex.test(dateString)) {
+		return false;
+	}
+	const date = new Date(`${dateString}T00:00:00.000Z`);
+	const timestamp = date.getTime();
+	if (isNaN(timestamp)) {
+		return false;
+	}
+	const [year, month, day] = dateString.split("-").map(Number);
+	return (
+		date.getUTCFullYear() === year &&
+		date.getUTCMonth() + 1 === month &&
+		date.getUTCDate() === day
+	);
```

Comment:
Why is this needed?

Reply:
When creating a Date object from input, it'll try to adjust the the date to something valid.
`new Date("02/31/2025")` becomes `3/3/2025` for example. This is just to check it matches user input.